src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/OrderingHelper.cs

```diff
@@ -0,0 +1,361 @@
+        /// </summary>
+        private static IProjectTree2 GetNextSibling(IProjectTree projectTree)
+        {
+            return GetSiblingByDisplayOrder(projectTree, (i, orderedChildren) =>
+            {
+                if (i == (orderedChildren.Count() - 1))
+                {
+                    return null;
+                }
+
+                return orderedChildren.ElementAt(i + 1) as IProjectTree2;
+            });
+        }
+
+        /// <summary>
+        /// Gets a sibling of the given project tree based on the move action. Can return null.
+        /// </summary>
+        private static IProjectTree GetSiblingByMoveAction(IProjectTree projectTree, MoveAction moveAction)
+        {
+            switch (moveAction)
+            {
+                case MoveAction.Up:
+                    return GetPreviousSibling(projectTree);
+
+                case MoveAction.Down:
+                    return GetNextSibling(projectTree);
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets a reference element based on the given project tree and move action. Can return null.
+        /// </summary>
+        private static ProjectItemElement GetReferenceElementByMoveAction(UnconfiguredProject unconfiguredProject, Project project, IProjectTree projectTree, MoveAction moveAction)
+        {
+            switch (moveAction)
+            {
+                case MoveAction.Up:
+                    return GetElements(unconfiguredProject, project, projectTree).FirstOrDefault();
+
+                case MoveAction.Down:
+                    return GetElements(unconfiguredProject, project, projectTree).LastOrDefault();
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Move a child element based on the reference element and move action.
+        /// </summary>
+        private static void MoveChildByMoveAction(ProjectItemElement element, ProjectItemElement referenceElement, MoveAction moveAction)
+        {
+            var parent = referenceElement.Parent;
+            if (parent != null)
+            {
+                parent.RemoveChild(element);
+
+                switch (moveAction)
+                {
+                    case MoveAction.Up:
+                        parent.InsertBeforeChild(element, referenceElement);
+                        break;
+
+                    case MoveAction.Down:
+                        parent.InsertAfterChild(element, referenceElement);
+                        break;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Moves child elements based on the reference element and move action.
+        /// </summary>
+        private static void MoveChildrenByMoveAction(IEnumerable<ProjectItemElement> elements, ProjectItemElement referenceElement, MoveAction moveAction)
+        {
+            // If we are moving elements down, we need to reverse order.
```

Comment:
*why*?

Reply:
If I don't reverse the elements before calling `InsertAfterChild`, the end result will reverse the ordering of the elements that were part of the folder.